pkg/apis/middleware/session.go

```diff
@@ -2,14 +2,31 @@ package middleware
 
 import (
 	"context"
+	"time"
 
 	"github.com/coreos/go-oidc"
 	sessionsapi "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
 )
 
+// ProxyState is not serializable and carries proxy-scoped values and settings.
+// It can, hypothetically, change between the provider calls
+type ProxyState struct {
```

Comment:
What's the benefit of this struct? We seem to be passing it into a lot of different places that don't really make much use of it?

Reply:
It carries common proxy state to be exposed to providers. Right now provider has no idea about what configuration is. We've discussed this here: https://github.com/oauth2-proxy/oauth2-proxy/pull/886#discussion_r516458097